iothub/service/src/Common/Exceptions/ErrorCode.cs

```diff
@@ -7,14 +7,24 @@
 namespace Microsoft.Azure.Devices.Common.Exceptions
 {
     /// <summary>
-    /// Error Codes for common IoT hub exceptions.
+    /// Error codes for common IoT hub exceptions.
     /// </summary>
     public enum ErrorCode
     {
 #pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+        /// <summary>
+        /// Used when the error code returned by the hub is unrecognized. If encountered, please report the issue so it can be added here.
```

Comment:
I hate that we return `InvalidErrorCode` when we don't know. Why don't we blindly pass the error code we receive back to the user? Seems we enforce a known filter, and that seems problematic to me. Ideally we'd have the code that was returned, and class with known string constants for users and other code.

Too late to change, I think. Open to ideas.

Reply:
Probably something we can add to the list of meaningful breaking changes we want to propose as part of a release like we did in Java?

Reply:
I'll start a list